experimenter/experimenter/changelog/views.py

```diff
@@ -1 +1,30 @@
-# Create your views here.
+from django.shortcuts import get_object_or_404
+from django.views.generic import ListView
+
+from experimenter.experiments.models import NimbusChangeLog, NimbusExperiment
+
+
+class NimbusChangeLogsView(ListView):
+    model = NimbusChangeLog
+    template_name = "changelog/test.html"
+    context_object_name = "experiment"
+
+    def get(self, request, *args, **kwargs):
+        self.object = self.get_object()
+        return super().get(request, *args, **kwargs)
+
+    def get_object(self, queryset=None):
+        slug = self.kwargs.get("slug")
+        experiment = get_object_or_404(NimbusExperiment, slug=slug)
+        return experiment
```

Comment:
here- You can eliminate the need for the `get_object` method by using the` get_queryse`t method provided by the ListView. By overriding the get_queryset method, you can filter the objects based on the slug and experiment, like 
```python
def get_queryset(self):
    slug = self.kwargs.get("slug")
    return NimbusChangeLog.objects.filter(experiment__slug=slug)

Reply:
and this also eliminates the need to manually fetch the NimbusExperiment object and simplifies the code.

Reply:
I did this so that we can add the experiment data to the context as well because it will be needed when implementing the timeline for the experiment right ? Please let me know if there's a better way of achieving this